build.assets/tooling/cmd/buf-plugin-ensure-paginated/run.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PLUGIN_NAME=
+
+if [ "$#" -lt 1 ]; then
+  echo "Usage: $0 <plugin-name> [args...]" >&2
+  exit 1
+fi
+
+if [[ $# -gt 0 ]]; then
+  PLUGIN_NAME="$1"
+  shift
+fi
+
+PLUGIN_DIR="${SCRIPT_DIR}/${PLUGIN_NAME}"
+if [ ! -d "${SCRIPT_DIR}/${PLUGIN_NAME}" ]; then
+  echo "ERROR: Plugin '$PLUGIN_NAME' not found." >&2
+  exit 1
+fi
+
+pushd "$PLUGIN_DIR" > /dev/null
```

Comment:
Haven't we already changed directory to `dirname $0`? How is `cd $0/$1` supposed to work? Do we actually need any of this? Can't we just `exec go -C "$(dirname -- "$0")" run . "$@"` with no additional argument, if this script is already in the directory of the package?